### PR TITLE
Fix Filters Directive Bad Logic

### LINKED ISSUE #174: Fix Bad Logic Issue With Filters
https://deepscan.io/dashboard/#view=project&pid=2129&bid=10806&subview=issues&impact=%5B%22Medium%22%5D&page=1

### PR DESCRIPTION
**Pull request by martypowell**
Resolves #174 

- Adds missing null check
- Fixes bad null check

### CODE REVIEW

File: js/apps/filter-page/directives/filters.js [hunk 1]
```diff
@@ -10,32 +10,35 @@
...
-						
-						$scope.activeFilters.forEach((filter) => {
-							if (!hasMatch) {
-								hasMatch = !!tags
-									.filter((tagName) => findFilterMatch(tagName, filter)).length;
-							}
-						});
-
-						filterFamily.isFilterActive = hasMatch;
+						if (filterFamily) {
```
Comment (ascire) [inline js/apps/filter-page/directives/filters.js]: Could be rewritten in order to maintain the same indentation level...

```
if (!filterFamily) {
    return;
}

// ...
```

File: js/apps/filter-page/directives/filters.js [hunk 2]
```diff
@@ -10,32 +10,35 @@
...
+							let hasMatch = false;
+
+							$scope.activeFilters.forEach((filter) => {
+								if (!hasMatch) {
+									hasMatch = !!tags
+										.filter((tagName) => findFilterMatch(tagName, filter)).length;
+								}
+							});
+
+							// This should probably be refactored to be immutable
```
Comment (ascire) [inline js/apps/filter-page/directives/filters.js]: You could start by moving the generation of `hasMatch` to a different function.

Comment (martypowell) [reply js/apps/filter-page/directives/filters.js]: @ascire I think thats a good idea. I wasn't going for a refactor here, this should be it's own method.

File: js/apps/filter-page/directives/filters.js (hunk 1 above)
Comment (martypowell) [reply js/apps/filter-page/directives/filters.js]: @ascire I like this better, changing.

**Comment by martypowell** [comment]
@ascire ready for another look.

**Pull request merged**